ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -459,8 +468,11 @@ private void initializeDisksAndReplicasOnNode(AmbryDataNode datanode, InstanceCo
             }
             ensurePartitionAbsenceOnNodeAndValidateCapacity(mappedPartition, datanode, replicaCapacity);
             // Create replica associated with this node.
+            HardwareState replicaState = diskState == HardwareState.UNAVAILABLE ? diskState
```

Comment:
Why does the `diskState` have to be taken into  account here? The `isDown()` method of `AmbryReplica` already evaluates `diskState`

Reply:
changes are made